File: city-weather-angular/src/app/services/weather.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs/internal/Observable';

@Injectable({
  providedIn: 'root'
})
export class WeatherService {
    //The service IP Address with which Angular will communicate. 
  private gatewayIp = "http://localhost:8080";
  private controller = "/city/weather";
  private url = this.gatewayIp.concat(this.controller);


  constructor(private http: HttpClient) { }

//This method is used to fetch weather-reports of all the cities.
public getWeatherReports():Observable<any>{

  return this.http.get<any>(this.url.concat('/all'));
}

//This method is used to fetch weather-reports of a city.
public getWeatherReport(cityId:string):Observable<any>{

  return this.http.get<any>(this.url.concat('/'+cityId));
}

}

File: city-service/src/main/java/com/mindfire/cityservice/service/WeatherService.java
package com.mindfire.cityservice.service;

import java.util.List;

import com.mindfire.cityservice.dto.CityWeatherReport;
import com.mindfire.cityservice.dto.Weather;

public interface WeatherService {


	/**
	 *  This method is used to fetch the Weather Report of a city.
	 *  
	 * @param cityId : The cityId which needs to be searched in database and return the cityName associated. 
	 * @return {@link Weather} : returns WeatherReport for a single city. 
	 */
	public Weather getWeatherReport(String cityId);
	
	
	/**
	 *  This method is used to fetch the Weather Report of all the cities in cityList
	 *  
	 * @return {@link List<CityWeatherReport>}
	 */
	public List<CityWeatherReport> getWeatherReports();


	

}
